client.go

```diff
@@ -5,13 +5,23 @@ import "github.com/kochavalabs/mazzaroth-xdr/xdr"
 // Client defines a mazzaroth client that exposes common
 // endpoints exposed by a mazzaroth readonly node.
 type Client interface {
-	TransactionSubmit(transaction xdr.Transaction) (*xdr.TransactionSubmitResponse, error)
-	TransactionLookup(transactionID xdr.ID) (*xdr.TransactionLookupResponse, error)
-	ReceiptLookup(transactionID xdr.ID) (*xdr.ReceiptLookupResponse, error)
-	BlockLookup(blockID xdr.Identifier) (*xdr.BlockLookupResponse, error)
-	BlockHeaderLookup(blockID xdr.Identifier) (*xdr.BlockHeaderLookupResponse, error)
-	AccountInfoLookup(accountID xdr.ID) (*xdr.AccountInfoLookupResponse, error)
-	ChannelInfoLookup(channelInfoType xdr.ChannelInfoType) (*xdr.ChannelInfoLookupResponse, error)
-	BlockHeightLookup() (uint64, error)
-	AbiLookup(channelID xdr.ID) (*xdr.Abi, error)
+	TransactionSubmitCall(channelID string, seed string, functionName string, parameters []string, nonce uint64, blockExpirationNumber uint64) (*xdr.Response, error)
+	TransactionSubmitContract(channelID string, seed string, contractBytes []byte, abiDef []byte, nonce uint64, blockExpirationNumber uint64) (*xdr.Response, error)
+	TransactionSubmitConfig(channelID string, seed string, owner string, nonce uint64, blockExpirationNumber uint64) (*xdr.Response, error)
+	TransactionUpdatePermission(channelID string, seed string, nonce uint64, blockExpirationNumber uint64, authorizedAddressStr string, alias string, authorizedAlias string, authorize bool) (*xdr.Response, error)
+	TransactionLookup(channelID string, transactionID string) (*xdr.Response, error)
+	TransactionLookupByBlockHeight(channelID string, blockHeight int) (*xdr.Response, error)
+	TransactionLookupByBlockID(channelID string, blockID string) (*xdr.Response, error)
+	ReceiptLookup(channelID string, transactionID string) (*xdr.Response, error)
+	// ReceiptListFromBlockHeight(channelID string, blockHeight int) (*xdr.Response, error)
+	// ReceiptListFromBlockID(channelID string, blockID string) (*xdr.Response, error)
+	BlockLookup(channelID string, transactionID string) (*xdr.Response, error)
+	BlockListFromBlockHeight(channelID string, blockHeight int) (*xdr.Response, error)
+	BlockListFromBlockID(channelID string, blockID string) (*xdr.Response, error)
```

Comment:
There is no query parameter for Block ID.  That is just the normal URL Parameter version which includes block id.  The Query Parameters for Blocks and Blockheaders are "height" and "number": https://github.com/kochavalabs/mazzaroth-docs-internal/blob/new-docs/api/public.md#list-get-v1channelschannel_idblocksnumberheight

Also, these can be used together in a query.  Height is the starting height and number the number of blocks to return.

Reply:
Changes for block and blockheaders list.